src/mainframe/models/schemas.py

```diff
@@ -16,6 +16,46 @@ class Error(BaseModel):
     detail: str
 
 
+class Package(BaseModel):
+    """Model representing a package queried from the database."""
+
+    scan_id: str
+    name: str
+    version: str = ""
+    status: str = ""
+    score: Optional[int] = 0  # Just for the sake of texting, remove upon merge
+    inspector_url: Optional[str] = ""
+    rules: list[str] = []
+    download_urls: list[str] = []
+    queued_at: str = ""
+    queued_by: str = ""
+    pending_at: Optional[str] = None
+    pending_by: Optional[str] = None
+    finished_at: Optional[str] = None
+    finished_by: Optional[str] = None
+    commit_hash: Optional[str] = ""
+
+    @classmethod
+    def build_from_db(cls, scan_result): # I'm... not sure of what type `scan_result` is. I'll ask later.
+        return cls(
+            scan_id=str(scan_result.scan_id),
+            name=scan_result.name,
+            version=scan_result.version,
+            status=scan_result.status,
+            score=scan_result.score,
+            inspector_url=scan_result.inspector_url,
+            rules=[rule.rule_name for rule in scan_result.rules],
+            download_urls=[url.url for url in scan_result.download_urls],
+            queued_at=scan_result.queued_at.isoformat(),
+            queued_by=scan_result.queued_by,
+            pending_at=scan_result.pending_at.isoformat() if scan_result.pending_at else None,
+            pending_by=scan_result.pending_by,
+            finished_at=scan_result.finished_at.isoformat() if scan_result.finished_at else None,
```

Comment:
I think these should be UNIX timestamps. The model can be typed as `Optional[int]`, or even better, be typed as `Optional[datetime]` with a custom serializer. The Pydantic docs actually has an exact example for this: https://docs.pydantic.dev/latest/concepts/serialization/#custom-serializers:
```py
class WithCustomEncoders(BaseModel):
    model_config = ConfigDict(ser_json_timedelta='iso8601')

    dt: datetime
    diff: timedelta

    @field_serializer('dt')
    def serialize_dt(self, dt: datetime, _info):
        return dt.timestamp()
```